compiler/rustc_codegen_llvm/src/llvm_util.rs

```diff
@@ -459,23 +459,25 @@ pub(crate) fn global_llvm_features(sess: &Session, diagnostics: bool) -> Vec<Str
     // Features that come earlier are overridden by conflicting features later in the string.
     // Typically we'll want more explicit settings to override the implicit ones, so:
     //
+    // * Features disabled by --target; are overridden by
     // * Features from -Ctarget-cpu=*; are overridden by [^1]
-    // * Features implied by --target; are overridden by
     // * Features from -Ctarget-feature; are overridden by
+    // * Features implied by --target; are overridden by
     // * function specific features.
     //
     // [^1]: target-cpu=native is handled here, other target-cpu values are handled implicitly
     // through LLVM TargetMachine implementation.
-    //
-    // FIXME(nagisa): it isn't clear what's the best interaction between features implied by
-    // `-Ctarget-cpu` and `--target` are. On one hand, you'd expect CLI arguments to always
-    // override anything that's implicit, so e.g. when there's no `--target` flag, features implied
-    // the host target are overridden by `-Ctarget-cpu=*`. On the other hand, what about when both
-    // `--target` and `-Ctarget-cpu=*` are specified? Both then imply some target features and both
-    // flags are specified by the user on the CLI. It isn't as clear-cut which order of precedence
-    // should be taken in cases like these.
```

Comment:
I don’t believe this change addresses this FIXME at all. It still isn’t at all clear to me, even if we agreed that we don’t want to allow some features to be disabled, that this specific way to build up the feature set is the absolutely the correct one.

For example:

```
rustc -Ctarget-cpu=native --target=x86_64-unknown-linux-gnu
```

pretty clearly shows that the intent is for the `-Ctarget-cpu=native` to take precedence over whatever `x86_64-unknown-linux-gnu` means (including it not having any of the fancy AVX extensions.)

On the other hand

```
rustc -Ctarget-cpu=pentium4 --target=x86_64v4-unknown-linux-gnu
```

is quite a bit more ambiguous – so do we want something that would run on a pentium4? Or something that would run on an x86_64v4 class hardware, which pentium4 is not? Should the ordering of flags be taken into the account (unusual for rustc, but standard behaviour in gcc/clang style CLIs.) Perhaps it would be better to just error outright?